contracts/src/v0.8/dev/interfaces/OCR2DRRegistryInterface.sol

```diff
@@ -0,0 +1,175 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.6;
+
+/**
+ * @title OCR2DR billing subscription registry interface.
+ */
+interface OCR2DRRegistryInterface {
+  struct RequestBilling {
+    address client;
+    // a unique subscription ID allocated by billing system,
+    uint64 subscriptionId;
+    // customer specified gas limit for the fulfillment callback
+    uint32 gasLimit;
+  }
+
+  /**
+   * @notice Get configuration relevant for making requests
+   * @return uint32 global max for request gas limit
+   * @return address[] list of registered DONs
+   */
+  function getRequestConfig() external view returns (uint32, address[] memory);
+
+  /**
+   * @notice Determine the charged fee that will be paid to the Registry owner
+   * @param data Encoded OCR2DR request data, use OCR2DRClient API to encode a request
+   * @param billing The request's billing configuration
+   * @return fee Cost in Juels (1e18) of LINK
+   */
+  function getRequiredFee(bytes calldata data, OCR2DRRegistryInterface.RequestBilling calldata billing)
+    external
+    returns (uint96);
+
+  /**
+   * @notice Estimate the execution cost in gas that will be reimbursed to the Node Operator who transmits the data on-chain
+   * @param billing The request's billing configuration
+   * @return gasAmount
+   */
+  function estimateExecutionGas(OCR2DRRegistryInterface.RequestBilling calldata billing) external returns (uint256);
+
+  /**
+   * @notice Estimate the total cost to make a request: gas re-imbursement, plus DON fee, plus Registry fee
+   * @param data Encoded OCR2DR request data, use OCR2DRClient API to encode a request
+   * @param billing The request's billing configuration
+   * @param donRequiredFee Fee charged by the DON that is paid to Oracle Node
+   * @return billedCost Cost in Juels (1e18) of LINK
+   */
+  function estimateCost(
```

Comment:
Should this function also take an expected return value as the fulfillment gas cost will change based on the # of bytes in the response?

Reply:
Maybe. See RegistryInterface comment.